delphi-frontend/src/main/java/au/com/integradev/delphi/cfg/api/Halt.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Sonar Delphi Plugin
+ * Copyright (C) 2025 Integrated Application Development
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package au.com.integradev.delphi.cfg.api;
+
+import java.util.Collections;
+import java.util.Set;
+
+/** A block which represents a {@code Halt}, an abrupt process termination */
+public interface Halt extends Block, Terminated {
```

Comment:
Should this implement `Terminus`, or is there a reason it doesn't?
If it did, we wouldn't need the `getSuccessors` override.